esphome/components/store_yaml/store_yaml.cpp

```diff
@@ -0,0 +1,88 @@
+#include "store_yaml.h"
+#include "esphome/core/log.h"
+#include "esphome/core/helpers.h"
+
+namespace esphome {
+namespace store_yaml {
+
+static const char *const TAG = "store_yaml";
+
+void StoreYamlComponent::dump_config() {
+  if (this->show_in_dump_config_) {
+    ESP_LOGCONFIG(TAG, "YAML:");
+    RowDecompressor dec(ESPHOME_YAML, ESPHOME_YAML_SIZE);
+    std::string row;
+    while (dec.get_row(row)) {
+      ESP_LOGCONFIG(TAG, "%s", row.c_str());
+    }
+  }
+}
```

Comment:
Add more config info independent of `show_in_dump_config_`

```suggestion
void StoreYamlComponent::dump_config() {
  ESP_LOGCONFIG(TAG, "YAML:");
  ESP_LOGCONFIG(TAG, "  Compressed size: %zu", ESPHOME_YAML_SIZE);
  ESP_LOGCONFIG(TAG, "  Web server url: %s", (this->web_server_ != nullptr) ? this->web_server_url_ : "not configured!");
  if (this->show_in_dump_config_) {
    ESP_LOGCONFIG(TAG, "  Decompressed content:");
    RowDecompressor dec(ESPHOME_YAML, ESPHOME_YAML_SIZE);
    std::string row;
    while (dec.get_row(row)) {
      ESP_LOGCONFIG(TAG, "    %s", row.c_str());
    }
  }
}
```